### PR TITLE
SBX-51: Implement API endpoint for creating a new email schedule

### PR DESCRIPTION
**Pull request by lgaleana-llm**
This PR implements the API endpoint for creating a new email schedule. It includes updates to the FastAPI application, the database module, and the tests. The tests have been fixed to mock the add_email_schedule function instead of the database session.

### Test Plan

pytest

### CODE REVIEW

File: database.py [hunk 1]
```diff
@@ -11,3 +12,10 @@ class EmailSchedule(Base):
     subject = Column(String)
     body = Column(String)
     scheduled_time = Column(DateTime)
+
+
+def add_email_schedule(email_schedule: EmailSchedule, session: Session = None):
```
Comment (lgaleana) [inline database.py:23]: There is nothing being really persisted here. Let's at least use a sqlite db.

File: main.py [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+    subject: str
+    body: str
+    scheduled_time: str
+
+
+@app.post("/email-schedules")
+def create_email_schedule(email_schedule_request: EmailScheduleRequest):
+    email_schedule = EmailSchedule(**email_schedule_request.dict())
+    add_email_schedule(email_schedule)
+    return {"message": "Email schedule created successfully."}
```
Comment (lgaleana) [inline main.py]: Return the email schedule object. Validate it with FastAPI's response_model.

**Review by lgaleana — CHANGES_REQUESTED** [review]

**Comment by lgaleana-llm** [comment]
Sure, we can use SQLite for persisting the data. Here's how we can do it:

1. **Update the database module**: We will use SQLAlchemy's `create_engine` function to create an engine for the SQLite database. We will also use the `sessionmaker` function to create a Session class that is bound to this engine. The `add_email_schedule` function will then use this Session class to create a new session.

2. **Update the tests**: We will use the `testing.postgresql` package to create a temporary SQLite database for the tests. We will also update the `test_create_email_schedule` function to use this temporary database.

Does this plan sound good to you?

File: database.py (hunk 1 above)
Comment (lgaleana) [reply database.py:17]: You're right, we should use a real database for persistence. I'll update the `database.py` file to use SQLite. I'll also update the `add_email_schedule` function to handle the database session properly. Let me do that.

Comment (lgaleana) [reply database.py:17]: You're right, the current implementation of the `add_email_schedule` function doesn't actually persist the data. It just adds the `EmailSchedule` object to the session and commits the session, but the session is not connected to any database.

To fix this, we can use SQLAlchemy's `create_engine` function to create an engine that is connected to a SQLite database. We can then use this engine to create a `Session` object that is connected to the database.

Here's how to do it:

database.py
```python
from sqlalchemy import create_engine
from sqlalchemy.orm import sessionmaker

engine = create_engine('sqlite:///email_schedules.db')
Session = sessionmaker(bind=engine)

def add_email_schedule(email_schedule: EmailSchedule, session: Session = None):
    if session is None:
        session = Session()
    session.add(email_schedule)
    session.commit()
```

In this updated code, `create_engine` is used to create an engine that is connected to a SQLite database named `email_schedules.db`. `sessionmaker` is then used to create a `Session` class that is bound to this engine. The `add_email_schedule` function uses this `Session` class to create a session that is connected to the database.

I will amend the PR with these changes.

**Review by lgaleana — APPROVED** [review]

**Pull request merged**